acmw-db-app/pages/forgotPassword.js

```diff
@@ -0,0 +1,89 @@
+import Head from 'next/head'
+import styles from '../styles/Reset.module.css'
+import React, { useState } from 'react';
+  
+class forgotPassword extends React.Component {
+    constructor() {
+    super();
+    this.state = {
+      input: {},
+      errors: {}
+    };
+     
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+     
+  handleChange(event) {
+    let input = this.state.input;
+    input[event.target.name] = event.target.value;
+  
+    this.setState({
+      input
+    });
+  }
+     
+  handleSubmit(event) {
+    event.preventDefault();
+  
+    if(this.validate()){
+        let input = {};
+        input["email"] = "";
+        this.setState({input:input});
+    }
+  }
+  
+  validate(){
+      let input = this.state.input;
+      let errors = {};
+      let isValid = true;
+
+      if (typeof input["email"] !== "undefined") {
+
+        var pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
+        if (!pattern.test(input["email"] || input["email"] == "")) {
```

Comment:
can you explain this line to me? I don't understand why you are passing a boolean value to the test function

Reply:
yeah I logged what this evaluates to and if the string is empty, the value "true" gets passed to test. Although this happens to work since the word "test" is not formatted as a valid email address, we should not be passing it. If the regex is written correctly an empty string should not pass pattern.test anyway so it seems like you could just pass in whatever input["email"] is provided it is not null or undefined